lightning/src/ln/channelmanager.rs

```diff
@@ -9881,7 +9884,7 @@ macro_rules! create_refund_builder { ($self: ident, $builder: ty) => {
 	/// [Avoiding Duplicate Payments]: #avoiding-duplicate-payments
 	pub fn create_refund_builder(
 		&$self, amount_msats: u64, absolute_expiry: Duration, payment_id: PaymentId,
-		retry_strategy: Retry, max_total_routing_fee_msat: Option<u64>
+		retry_strategy: Retry, route_params_config: Option<RouteParametersConfig>
```

Comment:
Given the default `RouteParametersConfig` is used if `None`, perhaps we should remove the `Option` from here and throughout? Drawback is that user call sites already passing `None` would need to be updated on upgrade. @TheBlueMatt not sure if you have a preference here.

Reply:
Thanks a lot for the pointer! My initial reasoning for using `Option` was:

1. It keeps the call site clean.  
2. Since users aren’t manually passing a route parameters config, I felt it was more intuitive to pass `None` rather than `RouteParametersConfig::new()`.

Would love to hear your and Matt’s thoughts on this!

Reply:
Updated the call-site in [pr3342.10](https://github.com/shaavan/rust-lightning/commits/pr3342.10), so now the user calls RouteParametersConfig::default() instead of None in default case
Thanks for the pointer!